ckanext/datastore/backend/postgres.py

```diff
@@ -1392,6 +1392,8 @@ def format_results(context, results, data_dict):
                                                  field['type'])
         records.append(converted_row)
     data_dict['records'] = records
+    if data_dict.get('records_truncated', False):
+        data_dict['records'] = data_dict['records'][:rows_max]
```

Comment:
I'm wondering, won't there be any performance penalty for big lists with ~32000 records? I tested it locally with mock data and it doesn't seem to be a problem, but if you have a real dataset, it would be great to confirm it.  Otherwise, just popping the last record will do the same job, as I understand? and, due to lists nature, it won't consume additional memory/processor time

Reply:
32000 comes from https://github.com/ckan/ckan/pull/3523#issuecomment-296304558 measured against a real dataset

Reply:
Nice thinking @smotornyuk! I tried it and `.pop()` is twice the speed of `[:rows_max]`
```
>>> import timeit
>>> t1 = timeit.Timer('a=500000*[\'a\'];a.pop()')
>>> t2 = timeit.Timer('b=500000*[\'b\'];b[:-1]')
>>> print(t1.timeit(1000)/1000, t2.timeit(1000)/1000)
(0.0023797738552093508, 0.004582674026489258)
```
I'll make the change